extension/packagemanagerextension/nbextension/tree.js

```diff
@@ -0,0 +1,68 @@
+define(function (require) {
+    var $ = require('jquery');
+    var Jupyter = require('base/js/namespace');
+    var utils = require('base/js/utils');
+    var scripts = require('./scripts');
+    var urls = require('./urls');
+    var methods = require('./methods');
+
+    function init(dir) {
```

Comment:
We should call the api to load the package information when the modal box opens. The extension should expose a function to create a modal box and fill it with content. Like in here: https://gitlab.cern.ch/swan/jupyter/blob/qa/SwanShare/js/modal.js#L167

Reply:
May I know why it is better than making the API call and getting the contents before opening the modal. I currently have the function to create a modal defined in common.js.

Reply:
You are calling it when the extension loads, correct? In Jupyter, the files list is loaded dynamically, without the need to reload the page. If you do not reload the page, you do not reload your extension therefore you don't update any of the information (and don't even get the information when you change Projects).

Reply:
And I would say it's good practice to give the user the most recent information we got, whenever he wants to see it.

Reply:
> You are calling it when the extension loads, correct? In Jupyter, the files list is loaded dynamically, without the need to reload the page. If you do not reload the page, you do not reload your extension therefore you don't update any of the information (and don't even get the information when you change Projects).

I'm calling the API each time the sidebar shows up. Please tell me if that is what we are looking forward to.

Reply:
That's what I want, yes.